src/NuGet.Core/NuGet.ProjectModel/PackageSpec.cs

```diff
@@ -239,8 +239,8 @@ public bool Equals(PackageSpec other)
                    EqualityUtility.DictionaryEquals(PackInclude, other.PackInclude, (s, o) => StringComparer.Ordinal.Equals(s, o)) &&
                    EqualityUtility.EqualsWithNullCheck(PackOptions, other.PackOptions) &&
 #pragma warning restore CS0612 // Type or member is obsolete
-                   EqualityUtility.OrderedEquals(Dependencies, other.Dependencies, dep => dep.Name, StringComparer.OrdinalIgnoreCase) &&
-                   EqualityUtility.OrderedEquals(TargetFrameworks, other.TargetFrameworks, tfm => tfm.TargetAlias, StringComparer.OrdinalIgnoreCase) &&
+                   EqualityUtility.OrderedEquals(Dependencies, other.Dependencies, (a, b) => StringComparer.OrdinalIgnoreCase.Compare(a.Name, b.Name)) &&
```

Comment:
There has been an engaging conversation on this pull request regarding performance improvements. However, I'm curious about the following:

1. Why are we invoking the EqualityUtility.OrderedEquals method in the first place?
2. Have we considered maintaining the collection in sorted order? In other words, using a `SortedList` instead of a `List` could avoid the need to invoke `OrderBy` or `Sort` methods later. From examining a few callers to the `EqualityUtility.OrderedEquals` method, the collection appears to be sorted only on one key, rather than projecting on different keys.